App/features/cafes/reducer.ts

```diff
@@ -1,82 +1,165 @@
 import { combineReducers } from 'redux'
+import { PersistConfig, persistReducer } from 'redux-persist'
+import { AsyncStorage } from 'react-native'
 import { ActionType, getType } from 'typesafe-actions'
 
 import * as actions from './actions'
+import { CafeSessionsData, Cafes, Cafe } from './models'
 
-export interface RegisterCafes {
-  readonly [peerId: string]: {
+export interface CafesState {
+  readonly cafes: Cafes
+  readonly cafeSessions: {
+    readonly sessions: CafeSessionsData
+    readonly processing: boolean
     readonly error?: string
   }
 }
 
-export interface DeregisterCafes {
-  readonly [id: string]: {
-    readonly error?: string
-  }
-}
```

Comment:
Consolidates the `RegisterCafes` and `DeregisterCafes` state into a single `Cafes` state that tracks the `registering`, `registered`, and `unregistering` state of any cafe. This data is also persistent so we can get the cafe tokens when we need them